### PR TITLE
Parallelize cert creation and test integration

### PR DESCRIPTION
**Pull request by smarterclayton**
@liggitt

### CODE REVIEW

**Comment by smarterclayton** [comment]
[test]

File: pkg/util/parallel/parallel.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+			if err := fns[i](); err != nil {
+				errCh <- err
+			}
+			wg.Done()
+		}(i)
+	}
+	wg.Wait()
+	close(errCh)
+	var errs []error
+	for err := range errCh {
```
Comment (liggitt) [inline pkg/util/parallel/parallel.go]: Didn't know you could range on a channel

File: hack/test-integration.sh [hunk 2]
```diff
@@ -26,6 +26,7 @@ function cleanup()
 package="${OS_TEST_PACKAGE:-test/integration}"
 tags="${OS_TEST_TAGS:-integration no-docker}"
 
+export GOMAXPROCS="$(grep "processor" -c /proc/cpuinfo 2>/dev/null || sysctl -n hw.logicalcpu 2>/dev/null || 1)"
```
Comment (liggitt) [inline hack/test-integration.sh]: No idea how well this works across the platforms we care about

Comment (smarterclayton) [inline hack/test-integration.sh]: Linux and then Mac

File: pkg/util/parallel/parallel.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/util/parallel/parallel.go]: Very useful in certain cases

File: hack/test-integration.sh (hunk 2 above)
Comment (liggitt) [inline hack/test-integration.sh]: why not a tiny go program we can capture the output of, e.g. http://play.golang.org/p/E_mkDidnca

**Comment by liggitt** [comment]
desire for bash avoidance, but LGTM otherwise

File: hack/test-integration.sh (hunk 2 above)
Comment (smarterclayton) [inline hack/test-integration.sh]: Eh... [merge] and we can clean it up later.  I kind of want to make our integration stuff a bit better but not now.

File: pkg/util/parallel/parallel.go (hunk 1 above)
Comment (derekwaynecarr) [inline pkg/util/parallel/parallel.go]: Does it check the chan is closed?  Or can you range and block on an open chan?

Comment (smarterclayton) [inline pkg/util/parallel/parallel.go]: Range exits when channel is closed and empty.  Although now I want to do a test to prove that...

**Comment by smarterclayton** [comment]
Reading this failure... we should have been forbidden to list cluster policies... but we were allowed?

**Comment by smarterclayton** [comment]
Fixing in #3033 instead

**Pull request closed without merging**